examples/angular/src/app/lib/services/dotcms-context/page-context.service.ts

```diff
@@ -0,0 +1,51 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable } from 'rxjs';
+
+import { isInsideEditor } from '@dotcms/client';
+
+import { DotCMSPageAsset, DynamicComponentEntity } from '../../models';
+
+export interface ComponentItem {
+  component: Promise<any>;
+  module?: Promise<any>;
+}
+
+export interface DotCMSPageContext extends DotCMSPageAsset {
+  isInsideEditor: boolean;
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class PageContextService {
+
+  private componentsMap!: Record<string, DynamicComponentEntity> ;
+  private pageContext = new BehaviorSubject<DotCMSPageContext | null>(null);
+  readonly pageContext$ = this.pageContext.asObservable() as Observable<DotCMSPageContext>;
+
+  get pageContextValue(): DotCMSPageContext {
+    return this.pageContext.value as DotCMSPageContext;
+  }
+
+  getComponentMap(): Record<string, DynamicComponentEntity> {
+    return this.componentsMap;
+  }
+
+  setComponentMap(components: Record<string, DynamicComponentEntity> ) {
+    this.componentsMap = components;
+  }
+
+  /**
+   * Set the context
+   *
+   * @protected
+   * @param {DotCMSPageAsset} value
+   * @memberof DotcmsContextService
+   */
+  setContext(value: DotCMSPageAsset) {
+    this.pageContext.next({
+      ...value,
+      isInsideEditor: isInsideEditor()
+    });
+  }
```

Comment:
The data is set on a router level, so it loads before the components in the library.

I'll create a `BehaviorSubject` because it's an Angular Standard and everyone can subscribe to any change. Also, they can access the latest sync value with this property

https://github.com/dotCMS/core/blob/14f6796f9813dc463f49629241b33247359d91c0/examples/angular/src/app/lib/services/dotcms-context/page-context.service.ts#L26-L28